### ISSUE TITLE
Is it must to include androidx.startup.InitializationProvider in your AndroidManifest.xml ?

### ISSUE THREAD

**Issue by bharu2** [xamarin/xamarin-android#7488]
### Android application type

Classic Xamarin.Android (MonoAndroid12.0, etc.)

### Affected platform version

VS2022 17.3.1

### Description

I have been updating my xamarin.androidx nugets. 

**Xamarin.AndroidX.Appcompat to 1.4.2.1 version.
Xamarin.Google.Android.Material to 1.6.1.1 version.**
It did installed its dependent packages. 
**I havent used these below two packages before in my project. They are just installed as part of updating Appcompat.
Xamarin.AndroidX.Lifecycle.Process
Xamarin.AndroidX.Startup.StartupRuntime** 
After this when I try to deploy my app to device, I am getting below exception.

**_Java.Lang.RuntimeException: 'Unable to get provider androidx.startup.InitializationProvider: androidx.startup.StartupException: androidx.startup.StartupException: java.lang.IllegalStateException: ProcessLifecycleInitializer cannot be initialized lazily. Please ensure that you have: _**
```
<meta-data
    android:name='androidx.lifecycle.ProcessLifecycleInitializer' 
    android:value='androidx.startup' />
```
**_under InitializationProvider in your AndroidManifest.xml'_**

So, I added the below piece of code, 
```
<provider android:name="androidx.startup.InitializationProvider" android:authorities="${applicationId}.androidx-startup" tools:node="merge">
      <meta-data android:name="androidx.lifecycle.ProcessLifecycleInitializer" android:value="androidx.startup" />
</provider>
```
After adding this the exception is gone, but 
What is the need of this piece of code in my manifest. Is it must include?
What are the impacts of adding this?

### Steps to Reproduce

Update Appcompat package to 1.4.2.1 in Android 12 targetted app.

**Comment by jpobst** [comment]
Ultimately, all Xamarin.Android does is provide generated wrappers that allow you to call Google’s AndroidX API. We do not have much expertise in how to use those API’s properly. 

You will likely need to use the Android API documentation and relevant forums to determine what the requirements are.

**Comment by gmck** [comment]
@jpobst 
I suggested @bharu2  post his problem here because those lines are automatically added to my merged manifest in the obj folders in my app and all of my test apps. 

I've never had to add any of those lines manually to my manifest in those apps, so I can only conclude they are automatically added via the build process by the presence of a particular androidx Nuget.  Therefore I would have thought that the build process is failing in some way for his particular app.

**Comment by jpobst** [comment]
Indeed, there is an `AndroidManifest.xml` inside `androidx.startup.startup-runtime.aar`:

![image](https://user-images.githubusercontent.com/179295/197663775-7494c291-eb7e-4901-a622-46add5d223f9.png)

@dellis1972 Are you familiar with how this works?

**Comment by dellis1972** [comment]
@jpobst manifestmerger should handle adding this automatically. Its default for .net 6+ but not for classic I think. 
So adding `<AndroidManifestMerger>manifestmerger.jar</AndroidManifestMerger>` to the csproj should help if they are using classic.

**Comment by gmck** [comment]
@dellis1972 , @jpobst 

 _Its default for .net 6+ but not for classic I think._ 

I'm using Classic, never even tried .net 6, and it appears to be the default with Classic as mentioned earlier. 

I don't have `<AndroidManifestMerger>manifestmerger.jar</AndroidManifestMerger>` in my csproj.  @bharu2 also states Classic in his post, so why doesn't it work for him?

**Comment by jpobst** [comment]
It looks like AndroidX packages will set it for you *if* you haven't already set it to something else:

**Xamarin.AndroidX.Startup.StartupRuntime.targets**

```xml
  <PropertyGroup>
    <AndroidManifestMerger Condition=" '$(AndroidManifestMerger)' == '' ">manifestmerger.jar</AndroidManifestMerger>
  </PropertyGroup>
```

Perhaps it has already been set in the `.csproj` or by another NuGet package?
